pennylane/optimize/shot_adaptive.py

```diff
@@ -33,33 +33,22 @@ class ShotAdaptiveOptimizer(GradientDescentOptimizer):
     In addition, weighted random sampling can be used to further distribute the
     shot budget across the local terms from which the Hamiltonian is constructed.
 
-    .. warning::
-
-        ``ExpvalCost`` is deprecated. Instead, it is recommended to simply
-        pass Hamiltonians to the :func:`~pennylane.expval` function inside QNodes.
-
-        .. code-block:: python
-
-            @qml.qnode(dev)
-            def ansatz(params):
-                some_qfunc(params)
-                return qml.expval(Hamiltonian)
+    .. note::
 
-        You should also set ``term_sampling='weighted_random_sampling'`` to continue using weighted
+        You should set ``term_sampling='weighted_random_sampling'`` to continue using weighted
         random sampling in ``ShotAdaptiveOptimizer``.

```

Comment:
Maybe we should remove this warning entirely, since I think the docstring of the `term_sampling` arg explains the weighted random sampling.